File: output/xCodeEval_remove_8000/C++/Go/2a503f72fb23a1b541ad35379533f609.go
package main

import (
	"fmt"
)

const (
	INF = int(1e9)
)

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func LIS(n int, a []int) int {
	b := make([]int, 0)
	b = append(b, a[0])
	for i := 1; i < n; i++ {
		if b[len(b)-1] < a[i] {
			b = append(b, a[i])
		} else {
			l, r := 0, len(b)
			for l < r {
				m := (l + r) / 2
				if b[m] < a[i] {
					l = m + 1
				} else {
					r = m
				}
			}
			b[r] = a[i]
		}
	}
	return len(b)
}

func main() {
	var n, m int
	fmt.Scan(&n, &m)
	c := make([][]int, n)
	for i := 0; i < n; i++ {
		c[i] = make([]int, m)
		for j := 0; j < m; j++ {
			fmt.Scan(&c[i][j])
		}
	}

	s := 0
	t := n*m + 1
	n = t + 1
	m = 2

	D := make([]int, n)
	hd := make([]int, n)
	suc := make([]int, 2*m)
	to := make([]int, 2*m)
	cap := make([]int, 2*m)
	cst := make([]int, 2*m)

	m = 0
	add_edge := func(x, y, c, w int) {
		suc[m] = hd[x]
		to[m] = y
		cap[m] = c
		cst[m] = w
		hd[x] = m
		m++
	}

	id11 := func(x, y, c, w int) {
		if w == 0 {
			suc[m] = hd[x]
			to[m] = y
			cap[m] = c
			cst[m] = 0
			hd[x] = m
			m++
			suc[m] = hd[y]
			to[m] = x

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Python/2a503f72fb23a1b541ad35379533f609.py
import sys

def add_edge(x, y, c, w):
    global m
    suc[m] = hd[x]
    to[m] = y
    cap[m] = c
    cst[m] = w
    hd[x] = m
    m += 1

def id11(x, y, c, w):
    if not w:
        add_edge(x, y, c, 0)
        add_edge(y, x, c, 0)
    else:
        add_edge(x, y, c, w)
        add_edge(y, x, c, w)

def spfa():
    global cz, op
    cz = 0
    op = 1
    inQ = [False] * n
    D = [float('inf')] * n
    F = [0] * n
    D[s] = 0
    F[s] = float('inf')
    while cz < op:
        u = Q[cz & mask]
        inQ[u] = False
        cz += 1
        for i in range(hd[u], -1, -1):
            v = to[i]
            c = cap[i]
            w = cst[i]
            if c and D[v] > D[u] + w:
                D[v] = D[u] + w
                F[v] = min(F[u], c)
                pre[v] = i
                if not inQ[v]:
                    Q[op & mask] = v
                    inQ[v] = True
                    op += 1
    return D[t] != float('inf')

def add_path():
    global flow, cost
    flow += F[t]
    u = t
    while u != s:
        i = pre[u]
        v = to[i ^ 1]
        f = F[t]
        c = cap[i]
        w = cst[i]
        cost += F[t] * w
        v = u
    flow += f
    c -= f

def id9():
    global flow, cost
    to[m] = s
    flow = 0
    cost = 0
    while spfa():
        add_path()
    return (cost, flow)

def main():
    n, m = map(int, input().split())
    c = []
    for _ in range(n):
        c.append(list(map(int, input().split())))
